lib/pymedphys/_trf/decode/table.py

```diff
@@ -12,197 +12,83 @@
-        rows = [
-            trf_table_contents[i : i + line_grouping]
-            for i in range(0, len(trf_table_contents), line_grouping)
-        ]
-
-        result = decode_table_data(rows, line_grouping)
-        tentative_state_codes = result[  # pylint: disable = unsubscriptable-object
-            :, linac_state_codes_column
-        ]
-
-        if set(tentative_state_codes).issubset(reference_state_codes):
-            decoded_results.append(decode_table_data(rows, line_grouping))
-            possible_column_adjustment_key.append(key)
-        else:
-            non_reference_state_codes = set(tentative_state_codes).difference(
-                reference_state_codes
-            )
-            non_reference_state_codes_found = non_reference_state_codes_found.union(
-                non_reference_state_codes
-            )
-
-    if not decoded_results:
-        raise ValueError(
-            "Decoded table didn't pass shape test. While attempting to "
-            "decode the TRF logfile there were some non-reference state "
-            "codes that were found. This may be the cause of this shape "
-            "test failure. The non-reference state codes found are "
-            f"{non_reference_state_codes_found}"
-        )
-
-    if len(decoded_results) > 1:
-        raise ValueError("Can't determine version of trf file from table shape")
-
-    decoded_rows = decoded_results[0]
-    column_adjustment_key = possible_column_adjustment_key[0]
-
-    return decoded_rows, column_adjustment_key
-
-
-def decode_rows_from_file(filepath):
-    with open(filepath, "rb") as file:
-        trf_contents = file.read()
-
-    header_length = determine_header_length(trf_contents)
-    trf_table_contents = trf_contents[header_length::]
-
-    decoded_rows, _ = decode_rows(trf_table_contents)
-
-    return decoded_rows
-
-
-def get_base_column_names():
-    return CONFIG["column_names"]
-
-
-def get_column_names(column_adjustment_key):
-    column_names = get_base_column_names()
-
-    if column_adjustment_key == "integrity_v3":
-        return column_names
-
-    filler_columns = [f"unknown{item}" for item in range(1, 5)]
-
-    column_names = filler_columns + column_names
-
-    if column_adjustment_key == "integrity_v4":
-        return column_names
-
-    if column_adjustment_key != "unity_experimental":
-        raise ValueError("Unexpected `column_adjustment_key`")
-
-    column_names = [item for item in column_names if "Dlg" not in item]
 
-    return column_names
 
+def decode_trf_table(trf_table_contents, header_table_contents):
 
-def decode_trf_table(trf_table_contents):
-    decoded_rows, column_adjustment_key = decode_rows(trf_table_contents)
+    version = header_table_contents["version"].values[0].astype(int)
+    item_parts_length = header_table_contents["item_parts_length"].values[0].astype(int)
+    item_parts = header_table_contents["item_parts"].values[0]
```

Comment:
Would it be worth doing this header type conversion back right after the header has been parsed. As in type fixing of the header contents is likely the job of the "decode header" function, not the "decode table" function.

Reply:
@SimonBiggs I'm not sure i understand. do you mean passing the variables (version, item_parts_length, item_parts) into the function decode_trf_table, instead of passing header_table_contents?

Reply:
I guess where I am confused is with the following piece:

`.astype(int)`

My thought is that when the `header_table_contents` is created, what if it was made int right away? This would mean that it wouldn't be the job of downstream functions to convert it to int.

I do actually believe you're doing this already, at least for `item_parts_length`:

https://github.com/pymedphys/pymedphys/blob/d4f9900404a71299dc1c8b90e5fc5bb28108727d/lib/pymedphys/_trf/decode/header.py#L162

Reply:
And potentially `.astype(int)` isn't needed for version either?

https://github.com/pymedphys/pymedphys/blob/d4f9900404a71299dc1c8b90e5fc5bb28108727d/lib/pymedphys/_trf/decode/header.py#L154

Reply:
I won't block merge on this though. Really just a "nit-pick". If you're happy I'll merge after the tests pass?